std/algorithm/searching.d

```diff
@@ -4426,6 +4427,429 @@ private void skipAll(alias pred = "a == b", R, Es...)(ref R r, Es es)
 }
 
 /**
+Returns a range with all occurrences of $(D substs) in $(R r)
+replaced with their substitution.
+
+Single value replacements (`'ö'.substitute!('ä', 'a', 'ö', 'o', 'ü', 'u)`) are
+supported as well and $(BIGO 1).
+
+Params:
+    r = an $(REF_ALTTEXT input range, isInputRange, std,range,primitives)
+    value = a single value which can be substituted in $(BIGO 1)
+    substs = a set of replacements/substitutions
+
+Returns: a range with the substitutions replaced.
+
+See_Also:
+$(REF replace, std, array) for an eager replace algorithm or
+$(REF translate, std, string) or $(REF tr, std, string)
+for string algorithms with translation tables.
+*/
+template substitute(substs...)
+if (substs.length >= 2 && isExpressions!substs)
+{
+    import std.range.primitives : ElementType;
+    import std.traits : CommonType;
+
+    static assert(!(substs.length & 1), "The number of substitutions parameters must be even");
+
+    /**
+      Substitute single values with Compile-Time substitution mappings.
+      Complexity: $(BIGO 1) thanks to D's $(D switch) guaranteeing O(1).
+    */
+    auto substitute(Value)(Value value)
+    if (!isInputRange!Value && !is(CommonType!(Value, typeof(substs[0])) == void))
+    {
+        switch (value)
+        {
+            import std.meta : aliasSeqOf;
+            import std.range : iota;
+            foreach (const i; aliasSeqOf!(iota(0, substs.length / 2)))
```

Comment:
I will never stop loving this pattern. Better be careful, though, as I think there's a hard limit of 256 cases allowed in switch statements (this bug may have been fixed though, not sure).